src/colmap/scene/image.h

```diff
@@ -83,6 +84,12 @@ class Image {
   inline const Rigid3d& CamFromWorld() const;
   inline Rigid3d& CamFromWorld();
 
+  // Camera to world pose prior.
+  inline const PosePrior& WorldFromCamPrior() const;
+  inline PosePrior& WorldFromCamPrior();
+  // Check if PosePrior is valid
+  inline bool HasPosePrior() const;
```

Comment:
Instead of adding a new field to `Image`, how about extending `DatabaseCache` with pose priors? `SetupPriorPoseFromDatabase` could then be part of `DatabaseCache` (update all priors in-place).

Reply:
I don't think so. I suggest that `DatabaseCache` exposes an `std::unordered_map<image_t, PosePrior>`, which the `IncrementalMapper` can forward to `PositionPriorBundleAdjuster::SetUpProblem`. This would also greatly simplify `SetupPriorPoseFromDatabase`.

Reply:
This looks great, thank you!